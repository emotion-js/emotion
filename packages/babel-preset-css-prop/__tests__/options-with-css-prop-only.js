import babelTester from 'babel-tester'
import preset from '@emotion/babel-preset-css-prop'

babelTester('options-with-css-prop-only', __dirname, {
  presets: [
    [
      preset,
      {
        sourceMap: false,
        cssPropOnly: true
      }
    ]
  ]
})
