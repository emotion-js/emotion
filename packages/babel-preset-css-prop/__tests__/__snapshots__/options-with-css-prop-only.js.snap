// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options-with-css-prop-only index 1`] = `
"import * as React from 'react'

export let Button = props => {
  return (
    <>
      <button
        css={{
          color: 'hotpink'
        }}
        {...props}
      />
    </>
  )
}


      ↓ ↓ ↓ ↓ ↓ ↓

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from 'react';
import { jsx as ___EmotionJSX } from \\"@emotion/core\\";
import { Fragment as ___Fragment } from \\"react\\";
var _ref = {
  name: \\"1q24rv0-Button\\",
  styles: \\"color:hotpink;;label:Button;\\"
};
export let Button = props => {
  return ___EmotionJSX(___Fragment, null, ___EmotionJSX(\\"button\\", _extends({
    css: _ref
  }, props)));
};"
`;

exports[`options-with-css-prop-only index.without.css 1`] = `
"import * as React from 'react'

export let Button = props => {
  return (
    <>
      <button
        {...props}
      />
    </>
  )
}


      ↓ ↓ ↓ ↓ ↓ ↓

import * as React from 'react';
export let Button = props => {
  return React.createElement(React.Fragment, null, React.createElement(\\"button\\", props));
};"
`;
